ui/components/ui/delineator/delineator.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { useCallback, useState } from 'react';
+import classnames from 'classnames';
+import {
+  BorderRadius,
+  AlignItems,
+  BackgroundColor,
+  IconColor,
+  Display,
+  JustifyContent,
+  FlexDirection,
+} from '../../../helpers/constants/design-system';
+import {
+  AvatarIcon,
+  Icon,
+  Box,
+  IconName,
+  IconSize,
+} from '../../component-library';
+import Preloader from '../icon/preloader/preloader-icon.component';
+import { DelineatorProps, DelineatorType } from './delineator.types';
+import { getIconPropsByType, overrideTextComponentColorByType } from './utils';
+
+const Loader = () => (
+  <div role="progressbar">
+    <Preloader size={16} />
+  </div>
+);
+
+/**
+ * Renders the icon on the right based on the loading state.
+ *
+ * @param options - Options bag
+ * @param options.isExpanded - Whether the delineator is expanded.
+ * @returns Either a loading spinner or an arrow icon.
+ */
+const ExpandableIcon = ({ isExpanded }: { isExpanded: boolean }) => {
+  return (
+    <Icon
+      name={isExpanded ? IconName.ArrowUp : IconName.ArrowDown}
+      size={IconSize.Sm}
+      color={IconColor.iconMuted}
+    />
+  );
+};
+
+const Header = ({
+  handleHeaderClick,
+  headerComponent,
+  iconName,
+  isCollapsible,
+  isExpanded,
+  isLoading,
+  type,
+}: {
+  handleHeaderClick: () => void;
```

Comment:
Minor, but I believe the standard is `onX` for the property, and `handleX` for the callback.

Reply:
Done